README.md

```diff
@@ -70,11 +70,8 @@ server.register(FastifySSEPlugin);
 server.get("/", function (req, res) {
     res.sse(
   (async function* () {
-    for await (const event of on(eventEmmitter, "update")) {
-      yield {
-        type: event.name,
-        data: JSON.stringify(event),
-      };
+    for await (const event of on(eventEmitter, "update")) {
```

Comment:
I don't think the original example works, unless I am misunderstanding how you emit the events.
The `event` variable is an Array (see example [here](https://nodejs.org/api/events.html#eventsonemitter-eventname-options)), so I'm not sure what `event.name` is expected to resolve to.

The example also introduces a memory leak since the AsyncIterator is never closed.  The documentation states you can listen to close events on `req.raw`, but it would be nice to include it in the example.  It's quite important.

Reply:
you could emit objects with `name` property but I see now what you mean.

This should fix it:
```suggestion
    for await (const [event] of on(eventEmitter, "update")) {
```

and emitting could be something like
```js
eventEmitter.emit("update", {name: "test", data: {}})
```